server/gokbg3/grails-app/services/org/gokb/FTUpdateService.groovy

```diff
@@ -448,36 +411,85 @@ class FTUpdateService {
+          result.tippTitle = kbc.title.getLogEntityId()
+          result.tippTitleName = kbc.title.name
+          result.tippTitleUuid = kbc.title.uuid
+          result.tippTitleMedium = kbc.title.medium.value
+          kbc.title.titleHistory?.each { he ->
+            if (he.date) {
+              def event = [:]
+              event.date = dateFormatService.formatIsoTimestamp(he.date)
+              event.from = []
+              if (he.from) {
+                event.from.addAll(he.from.collect { fe -> [id: fe?.id, uuid: fe?.uuid, name: fe?.name] })
+              }
+              event.to = []
+              if (he.to) {
+                event.to.addAll(he.to.collect { te -> [id: te?.id, uuid: te?.uuid, name: te?.name] })
+              }
+              event.id = he.id ?: ""
+              result.titleHistory.add(event)
             }
-            event.id = he.id ?: ""
-            result.titleHistory.add(event)
+          }
+          kbc.title.publisher?.each { pub ->
+            def publisher = [:]
+            publisher.name = pub.name ?: ""
+            publisher.id = pub.id ?: ""
+            publisher.uuid = pub.uuid ?: ""
+            result.titlePublishers.add(publisher)
+          }
+          kbc.title.variantNames.each { vn ->
+            result.altname.add(vn.variantName)
           }
         }
 
-        // publishers for all title types
-        result.titlePublishers = []
-        kbc.title?.publisher?.each { pub ->
-          def publisher = [:]
-          publisher.name = pub.name ?: ""
-          publisher.id = pub.id ?: ""
-          publisher.uuid = pub.uuid ?: ""
-          result.titlePublishers.add(publisher)
+        if (kbc.medium) result.medium = kbc.medium.value
+        if (kbc.status) result.status = kbc.status.value
+        if (kbc.publicationType) result.publicationType = kbc.publicationType.value
+
+        result.identifiers = []
+        kbc.getCombosByPropertyNameAndStatus('ids', 'Active').each { idc ->
+          result.identifiers.add([namespace    : idc.toComponent.namespace.value,
+                                  value        : idc.toComponent.value,
+                                  namespaceName: idc.toComponent.namespace.name])
         }
 
-        // prices for all title types
+        if (kbc.dateFirstOnline) result.dateFirstOnline = dateFormatService.formatIsoTimestamp(kbc.dateFirstOnline)
+        if (kbc.dateFirstInPrint) result.dateFristInPrint = dateFormatService.formatIsoTimestamp(kbc.dateFirstInPrint)
+        if (kbc.accessStartDate) result.accessStartDate = dateFormatService.formatIsoTimestamp(kbc.accessStartDate)
+        if (kbc.accessEndDate) result.accessEndDate = dateFormatService.formatIsoTimestamp(kbc.accessEndDate)
+        if (kbc.lastChangedExternal) result.lastChangedExternal = dateFormatService.formatIsoTimestamp(kbc.lastChangedExternal)
+
+        if (org.apache.commons.lang.StringUtils.isNotEmpty(kbc.publisherName)) result.publisherName = kbc.publisherName
```

Comment:
A simple check for `kbc.publisherName` should work here, no?

Reply:
yeah, works equally well, although the .isNotEmpty checks not only for NULL values, but also empty and only whitespace strings.

Reply:
Right, the whitespace-only checks would need `kbc.publisherName?.trim()` (which groovy will evaluate to `false` for empty/whitespace-only Strings)